src/coreclr/src/interop/comwrappers.cpp

```diff
@@ -732,28 +755,59 @@ void NativeObjectWrapperContext::Destroy(_In_ NativeObjectWrapperContext* wrappe
 {
     _ASSERTE(wrapper != nullptr);
 
+    // Check if the tracker object manager should be informed prior to being destroyed.
+    IReferenceTracker* trackerMaybe = wrapper->GetReferenceTracker();
+    if (trackerMaybe != nullptr)
+    {
+        // We only call this during a GC so ignore the failure as
+        // there is no way we can handle it at this point.
+        HRESULT hr = TrackerObjectManager::BeforeWrapperDestroyed(trackerMaybe);
+        _ASSERTE(SUCCEEDED(hr));
+        (void)hr;
+    }
+
     // Manually trigger the destructor since placement
     // new was used to allocate the object.
     wrapper->~NativeObjectWrapperContext();
     InteropLibImports::MemFree(wrapper, AllocScenario::NativeObjectWrapper);
 }
 
-NativeObjectWrapperContext::NativeObjectWrapperContext(_In_ void* runtimeContext, _In_opt_ IReferenceTracker* trackerObject)
+namespace
+{
+    // State ownership mechanism.
+    enum : int
+    {
+        TrackerObjectState_NotSet = 0,
+        TrackerObjectState_SetNoRelease = 1,
+        TrackerObjectState_SetForRelease = 2,
+    };
```

Comment:
nit: Use a named enum or enum class instead of an anonymous one.

Reply:
I am keeping this as is for .NET 5. Let's have these style conversations in the .NET 6 PR: https://github.com/dotnet/runtime/pull/45624.